src/meta/test/AuthProcessorTest.cpp

```diff
@@ -0,0 +1,326 @@
+        ASSERT_TRUE(code.ok());
+    }
+    // Alter a few attributes
+    {
+        cpp2::AlterUserReq req;
+        decltype(req.user_item) newUser;
+        newUser.set_account("user1");
+        newUser.set_is_lock(true);
+        req.set_user_item(std::move(newUser));
+        auto* processor = AlterUserProcessor::instance(kv.get());
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, resp.get_code());
+    }
+    // Test user does not exist
+    {
+        cpp2::AlterUserReq req;
+        decltype(req.user_item) newUser;
+        newUser.set_account("user");
+        req.set_user_item(std::move(newUser));
+        auto* processor = AlterUserProcessor::instance(kv.get());
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        ASSERT_EQ(cpp2::ErrorCode::E_NOT_FOUND, resp.get_code());
+    }
+    // Alter all attributes
+    {
+        cpp2::AlterUserReq req;
+        decltype(req.user_item) newUser;
+        newUser.set_account("user1");
+        newUser.set_max_queries_per_hour(10);
+        newUser.set_max_updates_per_hour(20);
+        newUser.set_max_connections_per_hour(30);
+        newUser.set_max_user_connections(40);
+        newUser.set_is_lock(false);
+        req.set_user_item(std::move(newUser));
+        auto* processor = AlterUserProcessor::instance(kv.get());
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, resp.get_code());
+    }
+    // Test GetUser
+    {
+        cpp2::GetUserReq req(FRAGILE, "user1");
+        auto* processor = GetUserProcessor::instance(kv.get());
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        decltype(resp.user_item) user;
+        user.set_account("user1");
+        user.set_max_queries_per_hour(10);
+        user.set_max_updates_per_hour(20);
+        user.set_max_connections_per_hour(30);
+        user.set_max_user_connections(40);
+        user.set_is_lock(false);
+        ASSERT_EQ(user, resp.get_user_item());
+    }
+}
+
+TEST(AuthProcessorTest, DropUserTest) {
+    fs::TempDir rootPath("/tmp/DropUserTest.XXXXXX");
+    std::unique_ptr<kvstore::KVStore> kv(TestUtils::initKV(rootPath.path()));
+    // Setup
+    {
+        auto code = TestUtils::createUser(kv.get(), false, "user1", "pwd",
+                                          false, 1, 2, 3, 4);
+        ASSERT_TRUE(code.ok());
+    }
+    // Simple drop.
+    {
+        cpp2::DropUserReq req(FRAGILE, "user1", false);
+        auto* processor = DropUserProcessor::instance(kv.get());
+        auto f = processor->getFuture();
+        processor->process(req);
+        auto resp = std::move(f).get();
+        ASSERT_EQ(cpp2::ErrorCode::SUCCEEDED, resp.get_code());
+    }
+
```

Comment:
You are right . In fact, this feature relates to other PR such as #400, #480, etc. this test scenario exists in another test case : `src/graph/test/UserTest.cpp` .  I need a final screening let eliminate duplication and complement deficiencies.

Reply:
> Thanks for your review @whitewum , let ours hold this PR review temporary, because this PR depends on other PR, there is too much duplicate code.
sure.